### PR TITLE
Add `ready` field for ticdc's status

### LINKED ISSUE #4263: Update the status of TiCDC
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, the following fields are included in the TiCDC status:
```
// TiCDCStatus is TiCDC status
type TiCDCStatus struct {
	Synced      bool                    `json:"synced,omitempty"`
	Phase       MemberPhase             `json:"phase,omitempty"`
	StatefulSet *apps.StatefulSetStatus `json:"statefulSet,omitempty"`
	Captures    map[string]TiCDCCapture `json:"captures,omitempty"`
}
```
And TiDB Operator checks the Pod status [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/ticdc_member_manager.go#L225) without considering if the Pod exists or not.

So we should:
1. Check if the Pod exists first and only add capture info if the Pod exist so that users are able to know if the Pod has been created or not
2. Add a new field `state` in the `type TiCDCCapture struct` to indicate whether the capture is ready or not.
 
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #4263 

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 

Apply Cluster and check status\

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/ticdc_member_manager.go [hunk 1]
```diff
@@ -221,23 +222,35 @@ func (m *ticdcMemberManager) syncTiCDCStatus(tc *v1alpha1.TidbCluster, sts *apps
...
+			capture.Ready = true
 		}
+
+		ticdcCaptures[podName] = capture
 	}
-	if len(ticdcCaptures) == int(tc.TiCDCDeployDesiredReplicas()) {
-		tc.Status.TiCDC.Synced = true
-	}
+
+	tc.Status.TiCDC.Synced = len(ticdcCaptures) == int(tc.TiCDCDeployDesiredReplicas()) && allCapturesReady
```
Comment (KanShiori) [inline pkg/manager/member/ticdc_member_manager.go:253]: Is it means that the operator collect status successfully if `Synced` is true?

Comment (DanielZhangQD) [reply pkg/manager/member/ticdc_member_manager.go:253]: `Synced: true` means that all cdc captures are ready.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: pkg/manager/member/ticdc_member_manager.go [hunk 2]
```diff
@@ -221,23 +222,35 @@ func (m *ticdcMemberManager) syncTiCDCStatus(tc *v1alpha1.TidbCluster, sts *apps
...
 
 	ticdcCaptures := map[string]v1alpha1.TiCDCCapture{}
+	allCapturesReady := true
 	for id := range helper.GetPodOrdinals(tc.Status.TiCDC.StatefulSet.Replicas, sts) {
 		podName := fmt.Sprintf("%s-%d", controller.TiCDCMemberName(tc.GetName()), id)
-		capture, err := m.deps.CDCControl.GetStatus(tc, int32(id))
+
+		_, err := m.deps.PodLister.Pods(tc.GetNamespace()).Get(podName)
+		if err != nil {
+			klog.Warningf("Failed to get Pod %s of [%s/%s], error: %v", podName, ns, tcName, err)
```
Comment (csuzhangxc) [inline pkg/manager/member/ticdc_member_manager.go:231]: do we need set `allCapturesReady = false` here? or L253 can handle it correctly?

**Review by csuzhangxc — APPROVED** [review]

File: pkg/manager/member/ticdc_member_manager.go (hunk 2 above)
Comment (KanShiori) [reply pkg/manager/member/ticdc_member_manager.go:231]: I think `len(ticdcCaptures) == int(tc.TiCDCDeployDesiredReplicas())`  can handle it

**Comment by DanielZhangQD** [comment]
Please also add UT cases and update existing e2e cases to cover the changes.

**Comment by KanShiori** [comment]
> Please also add UT cases and update existing e2e cases to cover the changes.

Done

**Comment by KanShiori** [comment]
/hold
e2e need update

**Comment by KanShiori** [comment]
> Please also add UT cases and update existing e2e cases to cover the changes.

Because of e2e case "upgrade 1.17 to the latest", the status check isn't compatible 😢

**Pull request merged**